tests/imports.rs

```diff
@@ -0,0 +1,82 @@
+use wasm::{validate, RuntimeError, RuntimeInstance, DEFAULT_MODULE};
+
+const UNMET_IMPORTS: &str = r#"
+(module
+    (import "env" "dummy1" (func (param i32)))
+    (import "env" "dummy2" (func (param i32)))
+    (func (export "get_three") (param) (result i32)
+        i32.const 1
+        i32.const 2
+        i32.add
+    )
+)"#;
+
+const SIMPLE_IMPORT_BASE: &str = r#"
+(module
+    (import "env" "get_one" (func $get_one (param) (result i32)))
+    (func (export "get_three") (param) (result i32)
+        call $get_one
+        i32.const 2
+        i32.add
+    )
+)"#;
+
+const SIMPLE_IMPORT_ADDON: &str = r#"
+(module
+    (func (export "get_one") (param) (result i32)
+        i32.const 1
+    )
+)"#;
+
+#[test_log::test]
+pub fn unmet_imports() {
+    let wasm_bytes = wat::parse_str(UNMET_IMPORTS).unwrap();
+    let validation_info = validate(&wasm_bytes).expect("validation failed");
+    let mut instance = RuntimeInstance::new(&validation_info).expect("instantiation failed");
+
+    let get_three = instance
+        .get_function_by_name(DEFAULT_MODULE, "get_three")
+        .unwrap();
+
+    assert_eq!(
+        RuntimeError::UnmetImport,
+        instance
+            .invoke::<(), i32>(&get_three, ())
+            .expect_err("Expected invoke to fail due to unmet imports")
+    );
+}
+
+#[test_log::test]
+pub fn compile_simple_import() {
+    let wasm_bytes = wat::parse_str(SIMPLE_IMPORT_BASE).unwrap();
+    let validation_info = validate(&wasm_bytes).expect("validation failed");
+    let mut instance =
+        RuntimeInstance::new_named("base", &validation_info).expect("instantiation failed");
+
+    let wasm_bytes = wat::parse_str(SIMPLE_IMPORT_ADDON).unwrap();
+    let validation_info = validate(&wasm_bytes).expect("validation failed");
+    instance.add_module("addon", &validation_info);
+
+    // assert_eq!((), instance.invoke_named("print_three", ()).unwrap());
+    // Function 0 should be the imported function
+    // assert_eq!((), instance.invoke_func(1, ()).unwrap());
+}
+
+#[test_log::test]
+pub fn run_simple_import() {
+    let wasm_bytes = wat::parse_str(SIMPLE_IMPORT_BASE).unwrap();
+    let validation_info = validate(&wasm_bytes).expect("validation failed");
+    let mut instance =
+        RuntimeInstance::new_named("base", &validation_info).expect("instantiation failed");
+
+    let wasm_bytes = wat::parse_str(SIMPLE_IMPORT_ADDON).unwrap();
+    let validation_info = validate(&wasm_bytes).expect("validation failed");
+    instance.add_module("env", &validation_info);

```

Comment:
This a bit nitpicky, but I am more in favor of an interface where if A depends on B's store values for instantiation and B depends on nothing, B is instantiated with a RuntimeInstance::new_named("B") and then we instantiate a with RuntimeInstance::new_named_with("A", instance_of_B/imports??) sort of thing, since instantiation of A might depend on already instantiated const values etc. in B (https://webassembly.github.io/spec/core/exec/modules.html#external-typing) here I think the spec necessitates that the imports are instantiated within their module store. For globals at least this is definitely required, since a global.get of an imported const global can be used in constant exprs during instantiation, and that itself needs to have been instantiated. I think in js module instantiation is also this way, but i am not sure. But this can stay too.